codex/slots/proofs/prover.nim

```diff
@@ -30,25 +30,27 @@ import ../types
 
 export backends
 
+logScope:
+  topics = "codex prover"
+
 type
-  AnyProof* = CircomProof
-  AnyInputs* = CircomInputs
-  AnyKeys* = CircomKey
-  AnyHash* = Poseidon2Hash
   AnyBackend* = CircomCompat
-  AnyBuilder* = Poseidon2Builder
+  AnyProof* = CircomProof
```

Comment:
Still unclear to me why we call those AnyXXX but I guess I can ask you in person when we chat. :-)

Reply:
Similar pattern is used in Nim with ints & floats with `SomeInteger = int | uint | int8 | ...` and `SomeFloat`.

Reply:
Guess it's like an "inverted" way of doing polymorphism, but at the type level instead of using inheritance.